admin/lib/solidus_admin/providers/main_nav.rb

```diff
@@ -8,20 +8,21 @@ module Providers
     Container.register_provider("main_nav") do
       start do
         container.namespace("main_nav") do
-          register("orders", MainNavItem.new(key: "orders", icon: "solidus_admin/inbox-line.svg", position: 10))
+          register("orders", MainNavItem.new(key: "orders", route: :orders_path, icon: "solidus_admin/inbox-line.svg", position: 10))
           register(
             "products",
             MainNavItem.new(
               key: "products",
+              route: :products_path,
               icon: "solidus_admin/price-tag-3-line.svg",
               position: 20
             )
-              .with_child(key: "option_types", position: 10)
+              .with_child(key: "option_types", route: :option_types_path, position: 10)
           )
-          register("promotions", MainNavItem.new(key: "promotions", icon: "solidus_admin/megaphone-line.svg", position: 30))
-          register("stock", MainNavItem.new(key: "stock", icon: "solidus_admin/stack-line.svg", position: 40))
-          register("users", MainNavItem.new(key: "users", icon: "solidus_admin/user-line.svg", position: 50))
-          register("settings", MainNavItem.new(key: "settings", icon: "solidus_admin/settings-line.svg", position: 60))
+          register("promotions", MainNavItem.new(key: "promotions", route: :promotions_path, icon: "solidus_admin/megaphone-line.svg", position: 30))
+          register("stock", MainNavItem.new(key: "stock", route: :stock_items_path, icon: "solidus_admin/stack-line.svg", position: 40))
+          register("users", MainNavItem.new(key: "users", route: :users_path, icon: "solidus_admin/user-line.svg", position: 50))
+          register("settings", MainNavItem.new(key: "settings", route: :stores_path, icon: "solidus_admin/settings-line.svg", position: 60))
```

Comment:
This setup is a bit hard to grok for me, I'm also not sure to understand the advantage of a fallback system over using different helpers for each menu item. Can you expand on it?

Reply:
Shouldn't we also consider that some routes will be available in both engines but controlled by a preference so that users may still stick to the old admin until they are ready to switch? I'm not sure what this adds to the mix here but I wanted to point it out in case it's relevant.

Reply:
> I want to hide our need to use two route proxies (spree & solidus_admin) until all the new admin has been done.

I thought it was better to keep it loud and clear, it's an existing and somewhat known pattern coming from rails itself, and we can refer to the original documentation instead of explaining how the proxy would work so that they can use it.

When it comes to the proxy I would propose to just evaluate it in the view_context (or the controller context) as that's the two natural environments in which urls are usually built:

```rb
register("users", MainNavItem.new(key: "users", route: -> { spree.users_path }, icon: "solidus_admin/user-line.svg", position: 50))
```

If I was a user that would look really straightforward and for us as developer we can skip a whole bunch of code.

Reply:
@kennyadsl we don't have an option yet for allowing users to select which sections they want to enable on the new admin (actually we barely have one section), so I think we should start with routes "hardcoded" in the proc.

The proc is flexible enough to accomodate whatever system we'll want to use for selecting the right route (a preference, cookies, a mix of both, etc.).

Whenever we'll add new sections we'll be able to also update the menu with new routes. At least until we setup a way to let the user cherry-pick the desired sections.

Reply:
After thinking about it, I agree it's better to require being explicit, as we still don't know how to implement all the stuff around the two types of routes.

Please, take a look at the last commit. Instead of providing the complete view context, I've limited it to the needed URL helper's proxies. It didn't feel well to me that a data object like `MainNavItem` had access to many things happening in the view context. In addition, this way, we don't need to worry about supporting weird stuff that could be done with the view context, and we don't need to worry about potential code injection in case somebody uses a database-backed main menu item.

Reply:
We don't have a ton of URLs yet, we always have time to change them in a separate PR that tackles exactly that problem, if any. This PR is already taking into account too many things, I think it's a wise choice to opt for this solution without anticipating too much. 👍